File: src/Animal.java
import java.util.ArrayList;

public class Animal {
    // variables
    static int idsAux=1;
    // atributos
    private int idAnimal;
    private String type;
    private String dateAdmission;
    private String birthdate;
    private double weight;
    private ArrayList<String> diseases = new ArrayList<>();
    private String feeding;
    private String feedingFrequencyHrs;
    private boolean vaccines;

    // constructor
    public Animal(String type, String dateAdmission, String birthdate, double weight, ArrayList<String> diseases, String feeding, String feedingFrequencyHrs, boolean vaccines){
        this.type = type;
        this.dateAdmission = dateAdmission;
        this.birthdate = birthdate;
        this.weight = weight;
        this.diseases = diseases;
        this.feeding = feeding;
        this.feedingFrequencyHrs = feedingFrequencyHrs;
        this.vaccines = vaccines;
        this.idAnimal = idsAux;
    }


    // getters - setters
    public int getIdAnimal() {
        return idAnimal;
    }
    public void setIdAnimal(int idAnimal) {
        this.idAnimal = idAnimal;
    }
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }
    public String getDateAdmission() {
        return dateAdmission;
    }
    public void setDateAdmission(String dateAdmission) {
        this.dateAdmission = dateAdmission;
    }
    public String getBirthdate() {
        return birthdate;
    }
    public void setBirthdate(String birthdate) {
        this.birthdate = birthdate;
    }
    public ArrayList<String> getDiseases() {
        return diseases;
    }
    public void setDiseases(ArrayList<String> diseases) {
        this.diseases = diseases;
    }
    public double getWeight() {
        return weight;
    }
    public void setWeight(double weight) {
        this.weight = weight;
    }
    public String getFeeding() {
        return feeding;
    }
    public void setFeeding(String feeding) {
        this.feeding = feeding;
    }
    public String getFeedingFrequencyHrs() {
        return feedingFrequencyHrs;
    }
    public void setFeedingFrequencyHrs(String feedingFrequencyHrs) {
        this.feedingFrequencyHrs = feedingFrequencyHrs;
    }
    public boolean getVaccines() {
        return vaccines;
    }
    public void setVaccines(boolean vaccines) {
        this.vaccines = vaccines;
    }

    // show data
    public void showData(){
        System.out.print("\nID: "+getIdAnimal()+"; Tipo de animal: "+getType()+"; Fecha de llegada: "+getDateAdmission()+"; Fecha de nacimiento: "+getBirthdate()+"; Peso: "+getWeight()+"; Enfermedades: ");
        for (int i = 0; i < getDiseases().size(); i++) {
            System.out.print(getDiseases().get(i));
            if (i+1 < getDiseases().size()){
                System.out.print(", ");
            }
            else {
                System.out.print("; ");
            }
        }
        System.out.print("Tipo de alimentación: "+getFeeding()+"; Frecuencia de alimentación por horas: "+getFeedingFrequencyHrs()+"; Vacunas: ");
        if (getVaccines() == true){
            System.out.print("Sí.\n");
        }
        else {
            System.out.print("No.\n");
        }
    }
}

File: src/Animal.py
class Animal:
    
    ids_aux = 1
    __id_animal = 0
    __type = ""
    __date_admission = ""
    __birthdate = ""
    __weight = 0
    __diseases = []
    __feeding = ""
    __feeding_frequency = ""
    __vaccines = False
    
    def __init__(self, type, date_admission, birthdate, weight, diseases, feeding, feeding_frequency, vaccines):
        self.__type = type
        self.__date_admission = date_admission
        self.__birthdate = birthdate
        self.__weight = weight
        self.__diseases = diseases
        self.__feeding = feeding
        self.__feeding_frequency = feeding_frequency
        self.__vaccines = vaccines
        self.__id_animal = ids_aux
        ids_aux = ids_aux + 1
    
    def showData(self):
        print(f"ID: {self.__id_animal}; Tipo de animal: {self.__type}; Fecha de llegada: {self.__date_admission}; Fecha de nacimiento: {self.__birthdate}; Peso:  {self.__weight}; Enfermedades: ")
        for x in self.__diseases:
            print(x)
        print(f"; Tipo de alimentación: {self.__feeding}; Frecuencia de alimentación: {self.__feeding_frequency}; Vacunas: ")
        if (self.__vaccines == True):
            print("Sí")
        else:
            print("No")
            
    # setters y getters   
    def get_id_animal(self):
        return self.__id_animal
    
    def get_type(self):
        return self.__type
    
    def get_date_admission(self):
        return self.__date_admission
    
    def get_birthdate(self):
        return self.__birthdate
    
    def get_weight(self):
        return self.__weight
    
    def get_diseases(self):
        return self.__diseases
    
    def get_feeding(self):
        return self.__feeding
    
    def get_feeding_frequency(self):
        return self.__feeding_frequency
    
    def get_vaccines(self):
        return self.__vaccines
    
    def set_type(self, value):
        self.__type = value
    
    def set_date_admission(self, value):
        self.__date_admission = value
    
    def set_birthdate(self, value):
        self.__birthdate = value
    
    def set_weight(self, value):
        self.__weight = value
    
    def set_diseases(self, value):
        self.__diseases = value
    
    def set_feeding(self, value):
        self.__feeding = value
    
    def set_feeding_frequency(self, value):
        self.__feeding_frequency = value
    
    def set_vaccines(self, value):
        self.__vaccines = value
    
    
    